data-otservbr-global/migrations/33.lua

```diff
@@ -1,3 +1,33 @@
 function onUpdateDatabase()
-	return false -- true = There are others migrations file | false = this is the last migration file
+	Spdlog.info("Updating database to version 34 (multi-world)")
+	db.query(
+		[[
+			CREATE TABLE IF NOT EXISTS `worlds`(
+			`id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
+			`name` varchar(32) NOT NULL,
+			CONSTRAINT `worlds_pk` PRIMARY KEY (`id`),
+			CONSTRAINT `worlds_unique` UNIQUE (`name`)
+		) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+		]]
+	)
+	db.query([[
+		ALTER TABLE players ADD COLUMN IF NOT EXISTS world_id int(11) unsigned not null default 1,
+		ADD CONSTRAINT `world_id_fk`
+        FOREIGN KEY (`world_id`) REFERENCES `worlds` (`id`)
+        ON DELETE CASCADE
+        ON UPDATE CASCADE
+		]])
+	
+	db.query([[
+		ALTER TABLE houses 
+		ADD COLUMN IF NOT EXISTS map_id int(25) not null,
+		ADD COLUMN IF NOT EXISTS world_id int(11) unsigned not null default 1,
+		ADD CONSTRAINT `houses_world_id_fk`
+		FOREIGN KEY (`world_id`) REFERENCES `worlds` (`id`)
+		ON DELETE CASCADE
+		ON UPDATE CASCADE
+		]])
```

Comment:
`ADD COLUMN` doesn't have a `IF NOT EXISTS` option
```suggestion
	db.query([[
		CREATE TABLE IF NOT EXISTS `worlds`(
			`id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
			`name` varchar(32) NOT NULL,
			CONSTRAINT `worlds_pk` PRIMARY KEY (`id`),
			CONSTRAINT `worlds_unique` UNIQUE (`name`)
		) ENGINE=InnoDB DEFAULT CHARSET=utf8;
	]])

	db.query([[
		ALTER TABLE players
			ADD COLUMN world_id int(11) unsigned not null default 1,
			ADD CONSTRAINT `world_id_fk` FOREIGN KEY (`world_id`) REFERENCES `worlds` (`id`)
			ON DELETE CASCADE
			ON UPDATE CASCADE
	]])

	db.query([[
		ALTER TABLE houses
			ADD COLUMN map_id int(25) not null,
			ADD COLUMN world_id int(11) unsigned not null default 1,
			ADD CONSTRAINT `houses_world_id_fk` FOREIGN KEY (`world_id`) REFERENCES `worlds` (`id`)
			ON DELETE CASCADE
			ON UPDATE CASCADE
	]])
```

Reply:
House tiles are still global, need to add world_id to that table and load the map accordingly <- Is this one necessary? The tile is already related to the HOUSE itself, and house has a world.id. It won't load on the wrong world since the one that calls the tile.load is the house, no?